codec/src/main/java/io/netty/handler/codec/base64/Base64.java

```diff
@@ -362,43 +362,63 @@ public boolean process(byte value) throws Exception {
+            final byte src3 = src[3];
+            if (src3 == EQUALS_SIGN) {
                 // Example: DkL=
-                int outBuff = (decodabet[src[0]] & 0xff) << 18 |
-                              (decodabet[src[1]] & 0xff) << 12 |
-                              (decodabet[src[2]] & 0xff) << 6;
-
+                final byte b1 = decodabet[src1];
                 // Packing bytes into a short to reduce bound and reference count checking.
                 if (dest.order() == ByteOrder.BIG_ENDIAN) {
-                    dest.setShort(destOffset, (short) ((outBuff & 0xff0000) >>> 8 | (outBuff & 0xff00) >>> 8));
+                    // The decodabet bytes are meant to straddle byte boundaries and so we must carefully mask out the
+                    // bits we care about.
+                    dest.setShort(destOffset, ((decodabet[src0] & 0x3f) << 2 | (b1 & 0xf0) >> 4) << 8 |
+                                              (b1 & 0xf) << 4 | (decodabet[src2] & 0xfc) >>> 2);
                 } else {
-                    dest.setShort(destOffset, (short) ((outBuff & 0xff0000) >>> 16 | (outBuff & 0xff00)));
+                    // This is just a simple byte swap of the operation above.
+                    dest.setShort(destOffset, (decodabet[src0] & 0x3f) << 2 | (b1 & 0xf0) >> 4 |
+                                              ((b1 & 0xf) << 4 | (decodabet[src2] & 0xfc) >>> 2) << 8);
                 }
                 return 2;
             }
 
             // Example: DkLE
-            final int outBuff;
             try {
-                outBuff = (decodabet[src[0]] & 0xff) << 18 |
-                          (decodabet[src[1]] & 0xff) << 12 |
-                          (decodabet[src[2]] & 0xff) <<  6 |
-                           decodabet[src[3]] & 0xff;
+                // Just directly set it as medium
+                if (dest.order() == ByteOrder.BIG_ENDIAN) {
+                    dest.setMedium(destOffset, (decodabet[src0] & 0x3f) << 18 |
+                                               (decodabet[src1] & 0xff) << 12 |
+                                               (decodabet[src2] & 0xff) << 6 |
+                                                decodabet[src3] & 0xff);
+                } else {
+                    final byte b1 = decodabet[src1];
+                    final byte b2 = decodabet[src2];
+                    // The goal is to byte swap the BIG_ENDIAN case above. There are 2 interesting things to consider:
+                    // 1. We are byte swapping a 3 byte data type. The top and the bottom byte switch, but the middle
+                    //    remains the same.
+                    // 2. The contents straddles byte boundaries. This means bytes will be pulled apart during the byte
+                    //    swapping process.
+                    dest.setMedium(destOffset, (decodabet[src0] & 0x3f) << 2 |
+                                            // The bottom half of b1 remains in the middle.
```

Comment:
then they will wrap onto the next line ... I think its just as clear this way and looks nicer IMHO ... wdyt ?